### PR TITLE
Graph leaky bucket

### PR DESCRIPTION
**Pull request by andre719mv**
Hi! 
I implemented Leaky Bucket for GraphQL. I am very limited in time so I did not formatted well my PR. Also I am pretty sure that you will not use my exact implementation. But if you will want to implement Leaky Bucket you may find something usefull here.

### CODE REVIEW

**Comment by nozzlegear** [comment]
Hey, thanks for the PR! Unfortunately I don't think I can merge it as is, because there's a lot of changes that go beyond just adding a leaky bucket to the graph service. For example, there's a lot of changes to the ShopifyService.cs file which comment out code and add some debug prints instead.

File: ShopifySharp/Services/ShopifyService.cs [hunk 1]
```diff
@@ -14,35 +14,39 @@ namespace ShopifySharp
 {
```
Comment (nozzlegear) [inline ShopifySharp/Services/ShopifyService.cs:14]: This file currently has too many unrelated changes for me to merge, particularly the debug logger printing.

**Comment by andre719mv** [comment]
Hey!
Sure. As I said before this PR is only to see the idea. I did not have time to prepare something that can be merged.

**Comment by nozzlegear** [comment]
Cool, thanks! I'll close the PR for now but will come back to look at it when we add leaky bucket support for the graph API.
